dandiapi/api/views/dandiset.py

```diff
@@ -101,6 +106,9 @@ def filter_queryset(self, request, queryset, view):
                     )
                 )
                 return queryset.order_by(ordering)
+            if ordering.endswith('stars'):
+                queryset = queryset.annotate(stars_count=Count('stars'))
+                return queryset.order_by(f"{'-' if ordering.startswith('-') else ''}stars_count")
```

Comment:
I find the original a bit hard to read
```suggestion
                queryset = queryset.annotate(stars_count=Count('stars'))
                prefix = '-' if ordering.startswith('-') else ''
                return queryset.order_by(f"{prefix}stars_count")
```